### ISSUE TITLE
Graphviz SVGs do not adapt to the column width

### ISSUE THREAD

**Issue by sjoelund** [sphinx-doc/sphinx#5002]
### Problem
- Graphviz SVGs do not adapt to the column width since they use the object tag instead of img

#### Procedure to reproduce the problem

The following produces a very wide image that is not resized (if SVG output is used in the HTML renderer):
```rst
.. digraph :: graphname

    ordering=out;
    ass1 [label="ast_assign"];
    ass2 [label="ast_assign"];
    ass3 [label="ast_assign"];
    ass4 [label="ast_assign"];
    elsif_l1 [label="ast_elsif_list"];
    elsif_l2 [label="ast_elsif_list"];
    id_a_1 [label="ast_id\na"];
    id_a_2 [label="ast_id\na"];
    id_a_3 [label="ast_id\na"];
    id_a_4 [label="ast_id\na"];
    id_b_1 [label="ast_id\nb"];
    id_b_2 [label="ast_id\nb"];
    id_b_3 [label="ast_id\nb"];
    id_b_4 [label="ast_id\nb"];
    id_c_1 [label="ast_id\nc"];
    id_c_2 [label="ast_id\nc"];
    id_c_3 [label="ast_id\nc"];
    id_c_4 [label="ast_id\nc"];
    i0 [label="ast_integer\n0"];
    i1 [label="ast_integer\n1"];

    ast_if -> ast_greaterthan;
    ast_if -> ass1;
    ast_if -> elsif_l1;
    ast_if -> ass4;

    ast_greaterthan -> id_a_1;
    ast_greaterthan -> id_b_1;

    ass1 -> id_c_1;
    ass1 -> id_a_2;

    elsif_l1 -> ast_eq;
    elsif_l1 -> ass2;
    elsif_l1 -> elsif_l2;
    ast_eq -> id_a_3;
    ast_eq -> id_b_2;
    ass2 -> id_c_2;
    ass2 -> i0;

    elsif_l2 -> ast_lessthan;
    elsif_l2 -> ass3;
    ast_lessthan -> id_a_4;
    ast_lessthan -> id_b_3;
    ass3 -> id_c_3;
    ass3 -> id_b_4;

    ass4 -> id_c_4;
    ass4 -> ast_uminus;
    ast_uminus -> i1;
```

#### Expected results
I would like an option to instead of creating:
```html
<object data="_images/xxx.svg" type="image/svg+xml">...</object>
```
I could create:
```html
<img alt="..." src="_images/graphviz-xxx.svg" />
```
Using the img-tag resizes the content properly. The only difference is that when using the object-tag, text cannot be copied.

Perhaps the following
```rst
.. digraph :: graphname
  :img-tag:
```
Sadly there is no option to change the output format only for a single graph (to png) either.

### Reproducible project / your project
- [small-project.zip](https://github.com/sphinx-doc/sphinx/files/2026614/small-project.zip)

### Environment info
- OS: Agnostic
- Python version: 3.6.5
- Sphinx version: 1.6.5, also checked the git master branch

**Comment by tk0miya** [comment]
I marked this as bug once, but I removed it now because this is not a bug.

Personally, I would not like to add options like `:img-tag:` to the directive. I think reST mark-ups should be independent with output format. The option is used only for HTML builds. It is a bit confusable.
But we have to render it more better. So I'll try to consider another solution later.

**Comment by sjoelund** [comment]
Yes, if it could be done without hacks like a new directive, that would be good as well. A global flag to change the behavior for HTML sort of works, but it also means text cannot be selected and links cannot be clicked in any Graphviz (at least in Chrome); I found this acceptable for my project.

**Comment by MarcusRiemer** [comment]
I am also stumbling over this and would much prefer to have texts in my graph to be selectable (therefor keeping the `object`-tag). My workaround consists of a single `CSS`-rule, namely `object { max-width: 100%; }`, that works fine in general but also is too broad and therefore affects non-graphviz `object`-elements. Would it be possible to add a specifc `graphviz` (or whatever unique name you see fitting) `CSS`-class to the `object` that is beeing generated so it is possible to target the embedded graphs specifically?

**Comment by tk0miya** [comment]
@MarcusRiemer Good idea! I just made a PR #5260 which implements your proposal :-)

**Comment by tk0miya** [comment]
I merged #5260. So all SVG images are automatically resized to 100% width.
Thank you all!
